Outreach and continuous learning (#33)
CaptainSifff/paper_teaching-learning-RSE: paper_teaching-learning-RSE.md

### Identifying skills and pathways

> As an area that generally requires a range of advanced skills,

Author: * this section should now look more balanced; extra paragraphs were added about:
   * competency frameworks developed in bioinformatics
   * an overview of domain-specific RSEs skill pathways (moved from the outlook)
* a new section on bioinformatics certification has been added after the HPC certification section
   * ideally, the certification chapter should grow independently of this one

> As an area that generally requires a range of advanced skills,

Reviewer: Great! we can sort this out with https://github.com/CaptainSifff/paper_teaching-learning-RSE/issues/15

> As an area that generally requires a range of advanced skills,

Reviewer 2: :+1:

research programme [@EXCALIBUR], is looking to understand and develop
training pathways to support the development of specialist skills in the HPC
and exascale domains. The project is gathering open source training materials
to develop curricula that support the training pathways that are underpinned
by high-quality training materials.

Bioinformatics is another field that actively works on competency frameworks.
The Bioinformatics Core Competencies [@Mulder2018; @Welch2016; @Welch2014],
BioExcel competency framework [@Matser2016], PerMedCoE competency
framework [@Lloret-Llinares2021], Research Data Management and Data
Stewardship competence framework [@Demchenko2021] and ELIXIR Data Stewardship
Competency Framework for Life Sciences [@Scholtens2019] eventually converged
into the EMBL-EBI Competency Hub [@CompetencyHub; @Lloret-Llinares2022],
where typical RSE and bioinformatician profiles at different levels
of seniority can be queried and compared against one another.
The Mastery Rubric for Bioinformatics [@Tractenberg2019] took a different
approach by identifying knowledge, skills and abilities instead
of competencies, and organising them in a taxonomy.

> 

Reviewer 2: There is some redundancy between 99-113 and the 3 following paragraphs. In particular the references are pretty much doubled in the same context, which seems weird. I suggest to Get rid of the "summary-listing" (104 - 109). And replace it with a somewhat extended version built from the following 3 paragraphs. The paragraph starting 120 belongs to the one above, it's both HPC. Can be shortened a bit.

Is it important to tell the history of the EMBL-EBI Competency Hub or would the info accessible from [@CompetencyHub; @Lloret-Llinares2022] also cover the info in Lines 128 - 133 sufficiently?

Author: Main points from today's online meeting:
* move EMBL Competency Hub to the appendix, where we can explain its full history
* move HPC certification to the appendix

The community also maintains the materials which are based around three core syllabuses --
Software Carpentry [@Wilson2006; @Wilson2016a], Data Carpentry [@Teal2015] and
Library Carpentry [@Baker2016; @Cope2018].
The training materials within these areas have been developed, reviewed and enhanced over several years ensuring that they represent best practice in training on these topics.
The core Carpentries lessons are targeted primarily at the beginner level.
However, the Carpentries Incubator [@CarpentriesIncubator] provides an environment for hosting additional community-developed training modules covering a wide range of other topics that have not gone through the peer review process of the core lessons.
The material in the Incubator increasingly includes more intermediate-level training modules.
After completion of a Carpentry workshop, learners can claim a certificate
of attendance [@CarpentriesLearnerCertificates].
Carpentry instructors must be certified before organizing a workshop.
There is a selection process [@CarpentriesBecomingInstructor] as well as
an instructor training course, upon which a certificate is delivered

> in the form of a digital badge [@Wilson2013a].

Reviewer 2: What is actually delivered is a certificate in the form of a PDF-page that looks like a certificate. I suggest to remove "digital badge" as it sounds a bit showy.

Author: Thanks! I could not find a picture of the instructor certificate, only that Carpentries refer to it as a "badge". I'll apply your suggestion.

### Identifying skills and pathways

The development of a standardized list of RSE competencies could help develop
metrics to measure an individual's progression in specific RSE skill sets,
e.g. using the Software Sustainability Institute's RSE Competencies
Toolkit [@RSECompetenciesToolkit2023].
These indicators would supplement the traditional grading system by providing
gamification elements that gratify learning and incentivise the discovery of
new skill sets, and thus contribute to reducing skill gaps in the RSE community.
Some indicators could be used in collaborative Web platforms to stimulate continuous learning

> or steer RSE behavior in a direction that would benefit all users of research software.

Reviewer 2: "Indicators" is not defined and also I think the word doesn#t apply to what is meant here ("competencies", badges). I don't think we should open the topic of experimental motivation enhancement for career development ("gamification"). I can't understand the last sentence about collaborative Web platforms and I also don't think there is consensus to establish instruments that "steer RSE behaviour".
I suggest to remove that paragraph.

Author: Agreed. Paragraph removed.

such as [Roberts1969], [Baxter2006], [Prlic2012], [Leprevost2014], [Wilson2014],
[Stodden2014], [Crusoe2016], [Crick2017], [Fehr2021], [Grossfield2022], some of
which are specifically tailored to group leaders, institutions and scientific
journal editors rather than RSEs [ChueHong2013], [ChueHong2014], [Katerbow2018], [Strasser2022].
Scientific journals have the advantage of reaching a large spectrum of
research scientists at all stages of their career.

Further ideas:
- teaching RSE infrastructure in relevant undergraduate courses of domain scientists, e.g. in statistics [Reinhart2021], [Beckman2021], [CetinkayaRundel2018]
- teach scientific software and research infrastructure early, e.g. in high school [Form2011], [Bain2020]
- adapting Carpentries and CodeRefinery material [ENCCS2022] for RSEs from industry and HPC facilities

> - localizing RSE teaching material and RSE information in languages other than English, e.g. [Astigarraga2022], [ClementFontaine2019b], [Appel2021], [Haim2021] (FIXME: find more examples of teaching material)

Author: * localized teaching material is certainly helpful to reach a wider audience
* look into ELIXIR for bioinformatics curricula (see #15)

Short primers on RSE skills, infrastructure and good coding practices
can be found in field-specific scientific articles and conference proceedings,
such as [@Roberts1969; @Baxter2006; @Prlic2012; @Leprevost2014; @Wilson2014c;
@Stodden2014; @Crusoe2016; @Crick2017; @Fehr2021; @Grossfield2022], some of
which are specifically tailored to group leaders, institutions and scientific
journal editors rather than RSEs [@ChueHong2013; @ChueHong2014; @Katerbow2018; @Strasser2022].
Scientific journals have the advantage of reaching a large spectrum of
research scientists at all stages of their career.

Localizing RSE teaching material and RSE information in languages other than English
can help reach a much wider audience by lowering the barrier to entry in the field.
In 2014 the community behind The Carpentries engaged in an international
effort to translate their training material in Spanish [@Wilson2014a],
Korean [@Wilson2014b] and Portuguese [@Silva2014].
Core lessons have been translated to Korean in 2015 [@Lee2015], and the Spanish

> core lessons are now officially part of the Software Carpentry [@CarpentriesSpanish].

Reviewer: ```suggestion
core lessons are now officially part of the Software Carpentry material[@CarpentriesSpanish].
```

Similarly, in the period 2015-2017, the Stack Overflow website launched localized
versions in Portuguese [@StackOverflowPortuguese], Russian [@StackOverflowRussian]
and Spanish [@StackOverflowSpanish] to reach a wider community.
There are also RSE short primers [@Astigarraga2022] and
RSE guidelines [@ClementFontaine2019b; @Appel2021; @Haim2021; @Schlauch2018b]
in non-English languages to address the need of specific communities.
FIXME: find more examples of teaching material in non-English languages.
FIXME: discuss the role of translations in overcoming linguistic and cultural barriers?

Teaching RSE in relevant undergraduate courses of domain scientists can be
the first point of entry in the field. For example, statistics curricula can
be used to showcase RSE infrastructure, e.g. the R programming language and
its ecosystem of statistics libraries and integrated development environments
[@Reinhart2021; @Beckman2021; @CetinkayaRundel2018].
There are also bioinformatics courses designed for high school students that

> covers topics such as pen-and-paper algorithm design, genomic database querying

Reviewer: ```suggestion
cover topics such as pen-and-paper algorithm design, genomic database querying
```

Author: The typos you reported were addressed. Thanks!

and data mining, and open data [@Form2011; @Bain2020].

Further ideas:

> - adapting Carpentries and CodeRefinery material [@ENCCS2022] for RSEs from industry and HPC facilities

Reviewer: This sounds like something we should discuss in a next meeting. Maybe @samumantha has an idea for that.

Some exemplary skills for which courses are already held are version control tools like git, HPC topics like multithreading, MPI and GPU computations, FAIR principles. 

##### Metrics and reward systems

The development of a standardized list of RSE competencies could help
develop metrics to measure an individual's progression in specific RSE
skill sets. These indicators would supplement the traditional grading
system by providing gamification elements that gratify learning and
incentivizes the discovery of new skill sets.
Some indicators could be used in GitHub/Lab to stimulate continuous learning or
steer RSE behavior in a direction that would benefit all users of research software.
Currently, code hosting platforms and software packaging services that allow
the public display of user activity are usually limited to basic statistics,
such as the number of commits or merge requests, but provide APIs to collect
this data and process it into more helpful metrics.

Digital badges [Fanfarelli2015], [McDaniel2016] are increasingly popular,
from personal badges rewarding contributions to a specific community,
or project badges highlighting coding practices, e.g. code coverage,

> continuous integration status, or code quality indicators.

Author: * personal badges could maybe help reduce skill gaps (fredo)
* look into [carpentry badges](https://software-carpentry.org/blog/2012/02/badges-finalized.html), which are not automated
* which institution would create RSE-badges?
   * badges could be based on certificates and get issued by certified institutions
* software badges serve entirely different purposes
   * code coverage badges and code maintainability/re-usability badges are actually useful for RSE grant proposals that require the software to be well tested and re-usable as a pre-requisite for funding
   * maybe move software badges to a self-contained paper about software policies and software quality assessment

The Fedora community developed a significant body of personal badges [FedoraBadges]
that reward contributions to open source software documentation, podcasts, conferences,
and code maintenance in the form of bugfixes, security fixes, code reviews and package releases.
The official GitHub Achievements [GitHubProfileAchievements] reward social behavior,
such as the number of merge requests, the number of commits with co-authors,
or the number of answered questions.
The GitHub user community also created unofficial trophies [GitHubProfileTrophy]
to highlight additional metrics, such as the number of repositories and opened tickets,
with a ranking system to introduce competition between users.

By default, platforms like GitHub and GitLab display the commit activity on users' profile pages,
in the form of a color-coded chart that measures the number of commits per day.
This statistic doesn't reflect the number of lines of code changed by a commit,
nor the nature of the change (bugfix, maintenance, documentation, etc.), and
can easily be gamed by automated tools [Rockstar], making its usefulness as a

> performance indicator debatable.

Author: * automatically assigned badges are easy to game
* badges can fade after a few years (diminishing usefulness or relevance)
* badges should not be used as performance indicators (for hiring and/or career progress)
* an argument could be made for having *less* metrics (GitHub allows users to hide their badges)

Author: @jcohen02 said:
> Potential badge use cases:
> - Show knowledge or skills in a particular area or at a particular level
> - Show code meets one or more specific best practice requirements
> - Highlight a particular level or length of experience in a particular topic...

journal editors rather than RSEs [@ChueHong2013; @ChueHong2014; @Katerbow2018; @Strasser2022].
Scientific journals have the advantage of reaching a large spectrum of
research scientists at all stages of their career.

Localizing RSE teaching material and RSE information in languages other than English
can help reach a much wider audience by lowering the barrier to entry in the field.
In 2014 the community behind The Carpentries engaged in an international
effort to translate their training material in Spanish [@Wilson2014a],
Korean [@Wilson2014b] and Portuguese [@Silva2014].
Core lessons have been translated to Korean in 2015 [@Lee2015], and the Spanish
core lessons are now officially part of the Software Carpentry material [@CarpentriesSpanish].
Similarly, in the period 2015-2017, the Stack Overflow website launched localized
versions in Portuguese [@StackOverflowPortuguese], Russian [@StackOverflowRussian]
and Spanish [@StackOverflowSpanish] to reach a wider community.
There are also RSE short primers [@Astigarraga2022] and
RSE guidelines [@ClementFontaine2019b; @Appel2021; @Haim2021] in non-English
languages to address the need of specific communities.

> Teaching RSE in relevant undergraduate courses of domain scientists can be
> the first point of entry in the field. For example, statistics curricula can

Reviewer 3: This can only provide a very broad overview for the domain scientists studies. E.g. at the TU Darmstadt, there is a course `Fundamentals of Digitalisation` in the first semester, that with 4 CP covers a lot of topics related to RDM and RSE, with quite ambitious/demanding goals. In the 5th semester there is a `Hands on tutorial digitalization` with 2 CP which ("only") demands software versioning and archiving.

Link to the Modulbuch: https://www.maschinenbau.tu-darmstadt.de/studieren/downloads/index.de.jsp#einleitung_bachelor_maschinenbau

```suggestion
Teaching RSE in relevant undergraduate courses of domain scientists can be the first point of entry in the field.
However, considering many RSEs come from domain studies, only fundamental concepts (of RDM and RSE) can be explained to and experienced by all of them. Those interested in computer-related work or even programming will gain the special knowledge in classes and projects they choose.
For example, statistics curricula can
```

PS: `computer-related work` please rephrase

Reviewer: Hi all, just to notify you I've merged some MRs to the master, which might lead to conflicts...

Some exemplary skills for which courses are already held are version control tools like git, HPC topics like multithreading, MPI and GPU computations, FAIR principles.

##### Metrics and reward systems

The development of a standardized list of RSE competencies could help
develop metrics to measure an individual's progression in specific RSE
skill sets. These indicators would supplement the traditional grading
system by providing gamification elements that gratify learning and
incentivizes the discovery of new skill sets, and thus contribute
to reducing skill gaps in the RSE community.

> Some indicators could be used in GitHub/Lab to stimulate continuous learning or

Reviewer: ```suggestion
Some indicators could be used used in collaborative Web platforms to stimulate continuous learning or
```

The GitHub user community also created unofficial trophies [@GitHubProfileTrophy]
to highlight alternative metrics, such as the number of repositories and opened tickets,
with a ranking system to introduce competition between users.

By default, platforms like GitHub and GitLab display the commit activity on users' profile pages,
in the form of a color-coded chart that measures the number of commits per day.
This statistic doesn't reflect the number of lines of code changed by a commit,
nor the nature of the change (bugfix, maintenance, documentation, etc.), and
can easily be gamed by automated tools [@Rockstar], making its usefulness as a
performance indicator debatable.
Yet, its ubiquity on personal pages and the lack of more objective metrics makes
it a de facto standard. The presence of this indicator can alter RSE behavior,
potentially in a counter-productive way since it priviledges quantity over quality.
For example, users can engage in contribution "streaking" to pad the chart [@Moldon2021] and
compete with other users via custom badges displaying their streaking performance [@GitHubStreakStats].

> 

Reviewer: I feel your text is well-written and well-researched and addresses a lot of interesting questions.
Nevertheless I wonder whether this level of detail is required for this paper or whether it is suited better for the separate paper on badges in #35 ... Currently we don't utilize the notion of badges outside of this chapter. If we give such a big textual weight to badges we must almost utilize it outside of this chapter, like in defining which badges could be used in the progression competency tables, or which ones are required for being a teacher.
I would also like to hear the opinions of some others on this @jcohen02 ? @jpthiele?

Author: Progress report:
- the off-topic paragraphs about commit charts and Fedora/GitHub badges have been removed
- the points discussed in https://github.com/CaptainSifff/paper_teaching-learning-RSE/issues/35#issuecomment-1607780631 have been added
- Ireland's PDF instructor badges and IBM badges have been added
